crates/predictor/src/predictor/dataset_reader.rs

```diff
@@ -10,7 +11,7 @@ struct UninitializedDataSetReader {
 }
 
 struct DataSetReader {
-    grib_readers: Vec<Box<GribReader>>
+    grib_readers: HashMap<i32, Box<GribReader>>
```

Comment:
This will have way higher memory usage and slightly higher lookup time. You should stick with the Vec, since hours will always be ints and we need every int populated anyway.